client/verta/verta/common/comparison/_verta_comparison.py

```diff
@@ -0,0 +1,84 @@
+# -*- coding: utf-8 -*-
+
+import abc
+import math
+
+from verta.external import six
+
+
+@six.add_metaclass(abc.ABCMeta)
+class _VertaComparison(object):
+    """
+    Base class for comparisons. Not for external use.
+
+    """
+
+    _OPERATOR = None  # variant of OperatorEnum
+    _SYMBOL = None  # e.g. "=="
+
+    def __init__(self, value):
+        self._value = value
+
+    def __repr__(self):
+        return "<comparison ({} {})>".format(
+            self._SYMBOL,
+            self.value,
+        )
+
+    @staticmethod
+    def isclose(a, b, rel_tol=1e-9, abs_tol=0):
+        if six.PY3:
+            # pylint: disable=no-member
+            return math.isclose(a, b, rel_tol=rel_tol, abs_tol=abs_tol)
+        else:
+            # equivalent implementation
+            # https://docs.python.org/3/library/math.html#math.isclose
+            return abs(a - b) <= max(rel_tol * max(abs(a), abs(b)), abs_tol)
+
+    @property
+    def value(self):
+        return self._value
+
+    def _operator_as_proto(self):
+        return self._OPERATOR
+
+    @classmethod
+    def _from_proto(cls, msg, value):
+        """
+        Returns an comparison object.
+
+        Parameters
+        ----------
+        msg : int
+            Variant of ``OperatorEnum``.
+        value : obj
+            Value to be compared against.
+
+        Returns
+        -------
+        :class:`_VertaComparison` subclass
+
+        """
+        for subcls in cls.__subclasses__():
+            if msg == subcls._OPERATOR:
+                return subcls(value)
+
+        raise ValueError("operator {} not recognized".format(msg))
+
+    @abc.abstractmethod
+    def compare(self, other_value):
```

Comment:
This is a case where we might think of implementing `__call__` instead of having this named `compare` method, because instances of `_VertaComparison` are basically functions which return boolean values according to the comparison. See the example ipython session below for what I'm talking about. I like this approach but if it feels overly-clever and not fluent we don't need to do this.

```
In [1]: class Comparison(object):
   ...:     def __init__(self, threshold):
   ...:         self.threshold = threshold
   ...:     def __call__(self, value):
   ...:         return self.threshold > value
   ...:

In [2]: comp = Comparison(0.5)

In [3]: comp(0.3)
Out[3]: True

In [4]: comp(0.7)
Out[4]: False
```

Reply:
I'd normally jump at opportunities to tap into Python's object model, but this case I'm wary about whether `__call__` offers a practical advantage, and `compare()` to me makes the API more explicit and therefore potentially readable.

I'd leave it open as an _option_ if we ever decide to allow, say, arbitrary callables as comparisons (which our current proto-based system wouldn't accommodate)